src/EPR.Payment.Service/Controllers/PaymentsController.cs

```diff
@@ -0,0 +1,76 @@
+﻿using Asp.Versioning;
+using EPR.Payment.Service.Common.Dtos.Request;
+using EPR.Payment.Service.Services.Interfaces;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.FeatureManagement.Mvc;
+
+namespace EPR.Payment.Service.Controllers
+{
+    [ApiVersion(1)]
+    [ApiController]
+    [Route("api/v{version:apiVersion}/payments")]
+    [FeatureGate("EnablePaymentsFeature")]
+    public class PaymentsController : ControllerBase
+    {
+        private readonly IPaymentsService _paymentsService;
+
+        public PaymentsController(IPaymentsService paymentsService)
+        {
+            _paymentsService = paymentsService ?? throw new ArgumentNullException(nameof(paymentsService));
+        }
+
+        [MapToApiVersion(1)]
+        [HttpPost]
+        [ProducesResponseType(typeof(Guid), 200)]
+        [ProducesResponseType(StatusCodes.Status400BadRequest)]
+        [ProducesResponseType(StatusCodes.Status500InternalServerError)]
+        [FeatureGate("EnablePaymentStatusInsert")]
+        public async Task<ActionResult<Guid>> InsertPaymentStatus([FromBody] PaymentStatusInsertRequestDto paymentStatusInsertRequest)
+        {
+            if (!ModelState.IsValid)
+            {
+                return BadRequest(ModelState);
+            }
+            try
+            {
+                var Id = await _paymentsService.InsertPaymentStatusAsync(paymentStatusInsertRequest);
+                return Ok(Id);
+            }
+            catch (ArgumentException ex)
+            {
+                return BadRequest(ex.Message);
+            }
+            catch (Exception ex)
+            {
+                return StatusCode(StatusCodes.Status500InternalServerError, $"Internal server error: {ex.Message}");
+            }
+        }
+
+        [MapToApiVersion(1)]
+        [HttpPut("{id}")]
+        [ProducesResponseType(200)]
```

Comment:
We are not returning anything so the return status code for a success should be;
[ProducesResponseType(204)]

Reply:
resolved